File: collections_management/collections_management/report/collections_report/collections_report.py
# Copyright (c) 2013, Element Labs and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe

def execute(filters=None):
	if not filters.site:
		filters.site = ""
	if not filters.collected_by:
		filters.collected_by = ""
	sqlq = """select creation,name,entry_type,machine_number,site,
			meter_reading,previous_reading,coins_expected,owner
		from `tabCollection Entry`
		where site like '%{}%'
		and owner like '%{}%'
		and docstatus = 1
		and creation BETWEEN '{}' AND '{}'
		""".format(filters.site,filters.collected_by,filters.from_date,filters.to_date)
	if filters.machine_number:
		sqlq+="""and machine_number = '{}'""".format(filters.machine_number)
	

	columns = [
		"Collected On:Datetime:150",
		"Bag No.:Link/Collection Entry:100",
		"Entry Type ::100",
		"Machine No.:Link/Asset:100",
		"Site:Link/Location:100",
		"Meter Reading::100",
		"Previous Reading::100",
		"Expected Coins:Currency:100",
		"Collected By::100"
	]

	data = frappe.db.sql(sqlq,as_list=1)
	
	return columns, data

File: collections_management/collections_management/report/collections_report/collections_report.js
// Copyright (c) 2016, Element Labs and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Collections Report"] = {
	"filters": [
		{
            "fieldname"	:"machine_number",
            "label"		: __("Machine Number"),
            "fieldtype"	: "Link",
            "options"	: "Asset"            
        },
        {
        	"fieldname"	:"site",
        	"label"		: __("Site"),
        	"fieldtype"	: "Link",
        	"options"	: "Warehouse"
        },
        {
        	"fieldname"	:"from_date",
        	"label"		: __("From Date"),
        	"fieldtype"	: "Datetime",
        	"reqd"		: 1,
        	"default"	: frappe.datetime.add_months(frappe.datetime.get_today(), -1)
        },
        {
        	"fieldname"	:"to_date",
        	"label"		: __("To Date"),
        	"fieldtype"	: "Datetime",
        	"reqd"		: 1,
        	"default"	: frappe.datetime.get_today()
        },
        {
        	"fieldname"	:"collected_by",
        	"label"		: __("Collection Agent"),
        	"fieldtype"	: "Link",
        	"options" 	: "User" 
        }
	]
}
